### ISSUE TITLE
Focus is lost and notebook disappears

### ISSUE THREAD

**Issue by iJorgen** [laurent22/joplin#10194]
### Operating system

Windows

### Joplin version

3.0.2

### Desktop version info

Joplin 3.0.2 (prod, linux)

Sync Version: 3
Profile Version: 46
Keychain Supported: No

Revision: a59ad20

Backup: 1.4.0

### Current behaviour

1. Select "All notes" in the left pane.

![Screenshot from 2024-03-24 09-03-15](https://github.com/laurent22/joplin/assets/52280788/816417dd-87bd-4155-9af3-aa9247c49887)

2. Select a note and then type any letter in the note. The focus is lost, the notebook is not shown in the list anymore and the second notebook in the list is shown.

It only happens when "All notes" is selected and not an individual notebook.

### Expected behaviour

Keep focus in the note and show the notebook on top in the list.

**Comment by graphit0** [comment]
Just noticed this issue happening by pressing backspace (instead of typing any letter) as well

ubuntu 22.04 and windows 11

Joplin 3.0.2

**Comment by laurent22** [comment]
@HahaBill, this bug was caused by your commit: 7638bdf171ead3b8ac268bf79d9f2275af6ecb42

**Comment by rcprcp** [comment]
This odd behavior also happens when searching before editing. 

* Use the search field. 
* Ensure that the are multiple notes in the results.  
* Select a note from the results. 
* Click into the note's body; type just one character into the selected note. (The issue does not seem to happen if you update the note's title.)  The UI updates, and you're viewing a different note.  It seems to be one from the list, but not the originally selected one. 

Operating system
Mac Os 14.3.1

Joplin version
Joplin 3.0.2 (prod, darwin)

Sync Version: 3
Profile Version: 46
Keychain Supported: Yes

Joplin is awesome, thank you!

**Comment by HahaBill** [comment]
Hi all, thank you for reporting the bug and I believe I caused this issue by introducing the `Toggle own sort order` feature for the `All notes` folder. 

We are trying to resolve this bug in PR #10222 by @AliceHincu, where she discovered the root of the issue and managed to fix it! We are still currently working on some code fixes.

**Comment by HahaBill** [comment]
> This odd behavior also happens when searching before editing.
> 
> * Use the search field.
> * Ensure that the are multiple notes in the results.
> * Select a note from the results.
> * Click into the note's body; type just one character into the selected note. (The issue does not seem to happen if you update the note's title.)  The UI updates, and you're viewing a different note.  It seems to be one from the list, but not the originally selected one.
> 
> Operating system Mac Os 14.3.1
> 
> Joplin version Joplin 3.0.2 (prod, darwin)
> 
> Sync Version: 3 Profile Version: 46 Keychain Supported: Yes
> 
> Joplin is awesome, thank you!

@rcprcp I tried to run it with a new fixed code, but the issue you mentioned persists. It might be a good idea to post the issue so someone else can work on the bug. The problem might be in the search functionality logic.

Also, since `All notes` is no longer focused when clicking and typing on a search bar, my intuition tells me that the problem might lie again in switching between the `SmartFilter` and `Folder` types.

But yeah, I agree that Joplin is awesome!

**Comment by personalizedrefrigerator** [comment]
See #10222

**Comment by personalizedrefrigerator** [comment]
According to a `git bisect`, 7638bdf171ead3b8ac268bf79d9f2275af6ecb42 is the first commit with this issue.

<details><summary>Notes</summary>

The bisect was done using `git bisect run` by applying the following diff, then running the test it adds:
```diff
diff --git a/packages/app-desktop/integration-tests/main.spec.ts b/packages/app-desktop/integration-tests/main.spec.ts
index 0bab7bc70..831c920d7 100644
--- a/packages/app-desktop/integration-tests/main.spec.ts
+++ b/packages/app-desktop/integration-tests/main.spec.ts
@@ -280,5 +280,35 @@ test.describe('main', () => {
 
 		await electronApp.close();
 	});
+
+	test('should allow editing in all-notes', async ({ mainWindow }) => {
+		const mainScreen = new MainScreen(mainWindow);
+		await mainScreen.waitFor();
+
+		const sidebar = mainWindow.locator('.rli-sideBar');
+		await expect(sidebar).toBeAttached();
+
+		await mainScreen.createNewNote('myTestNote');
+		const testNote = mainScreen.noteListContainer.getByText('myTestNote');
+		await expect(testNote).toBeAttached();
+
+		const allNotes = mainWindow.getByText('All notes');
+		await allNotes.click();
+		await new Promise(resolve => setTimeout(resolve, 2500));
+
+		await expect(mainWindow.locator(':focus')).toHaveText('All notes');
+
+		await expect(testNote).toBeAttached();
+
+		await testNote.click();
+
+		await mainScreen.noteEditor.focusCodeMirrorEditor();
+
+		await mainWindow.keyboard.type('Test...');
+
+		await new Promise(resolve => setTimeout(resolve, 4000));
+
+		await expect(testNote).toBeAttached();
+	});
 });
```

To make the test work, `lineHeight: 0` was removed from `Sidebar.tsx` (with `sed -i 's/lineHeight: 0//' packages/app-desktop/gui/Sidebar/Sidebar.tsx`).


</details>

**Comment by Julrob199** [comment]
Hope this gets fixed soon! This bug still exists in 3.0.7, it makes the app barely usable since weeks. Hope this i prioritized in the next update. I have the problem on the work windows computer and also here on my private linux. Thank you!!

This problem is not closed for me.

There are duplicates of many different types, which somehow all refer to the same problem: 
#10388 
#10236 
#10219
